docs/en/deploy/Running-Alluxio-On-Kubernetes.md

```diff
@@ -1689,4 +1689,137 @@ If this is the case, set the following properties to limit the path length:
 > Note: You may see performance degradation due to lack of node locality.
 
   {% endcollapsible %}
+  {% collapsible Worker Pods get OOMKilled by the Kubernetes scheduler %}
+This is most likely caused due to the Kubernetes configured
+[Pod resource limits](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits)
+having the `limits.memory` set too low.
+
+Firstly, double check the configured values for your Alluxio worker Pod `limits.memory`.
+**Note that the Pod consists of two containers, each with their own resource limits.**
+- If you used the Helm chart, the defaults are:
+
+```
+worker:
+  resources:
+    limits:
+      cpu: "4"
+      memory: "4G"
+    requests:
+      cpu: "1"
+      memory: "2G"
+
+jobWorker:
+  resources:
+    limits:
+      cpu: "4"
+      memory: "4G"
+    requests:
+      cpu: "1"
+      memory: "1G"
+```
+
+- You may also check the configured resource requests and limits using `kubectl describe pod`,
+`kubectl get pod`, or equivalent Kube API requests. eg.,
+
+```
+$ kubectl get po -o json alluxio-worker-xxxxx | jq '.spec.containers[].resources'
+{
+  "limits": {
+    "cpu": "4",
+    "memory": "4G"
+  },
+  "requests": {
+    "cpu": "1",
+    "memory": "2G"
+  }
+}
+{
+  "limits": {
+    "cpu": "4",
+    "memory": "4G"
+  },
+  "requests": {
+    "cpu": "1",
+    "memory": "1G"
+  }
+}
+```
+
+Next, ensure that the nodes that the Alluxio worker pods are running on have
+sufficient resources matching your configured values. You can check that the nodes
+you intend to schedule Alluxio worker Pods on have sufficient resources to meet
+your requests using `kubectl describe node`, `kubectl get node`, or equivalent Kube API requests. eg.,
+```
+$ kubectl get no -o json k8sworkernode-0 | jq '.status.allocatable'
+{
+  "cpu": "8",
+  "ephemeral-storage": "123684658586",
+  "hugepages-1Gi": "0",
+  "hugepages-2Mi": "0",
+  "memory": "64886128Ki",
+  "pods": "110"
+}
+```
+
+Isolating Alluxio worker Pods from other Pods in your Kubernetes cluster can be accomplished
+with the help of [node selectors](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector)
+and [node taints + tolerations](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/).
+- Keep in mind that the Alluxio worker Pod definition uses a
+  [DaemonSet](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/),
+  so there will be worker Pods assigned to all eligible nodes
+
+Lastly, ensure that the Alluxio workers' configured ramdisks (if any) do not exceed
+the Pod limits. See [the list of Alluxio configuration properties]({{ '/en/reference/Properties-List.html' | relativize_url }})
```

Comment:
I think making the ramdisk size < worker pod memory limit is not enough. The worker pod spends memory on:
1. The worker process' data structure like all the block location metadata
2. Buffers for I/O, both on and off heap
3. Other processes in the worker container

1 and 2 will be defined in the worker JVM properties. Does it make sense to mention the below?
```
worker Pod memory limit > worker JVM heap size + worker JVM off-heap size + ramdisk size
```
```

Reply:
Right, I wasn't sure exactly what else the Alluxio worker uses memory for so I wasn't able to specify beyond just the ramdisk itself. I think it does make sense to include those here as well.

Reply:
@jiacheliu3 Can you tell me what those JVM properties are for 1 and 2? `-Xms` and `-Xmx` are for the worker JVM heap min/max sizes, but what about off-heap size?